WordPress/Classes/Services/MediaCoordinator.swift

```diff
@@ -57,6 +58,32 @@ class MediaCoordinator: NSObject {
         uploadMedia(media)
     }
 
+    /// Cancels any ongoing upload of the Media and deletes it.
+    ///
+    /// - Parameter media: the object to cancel and delete
+    ///
+    func cancelUploadAndDeleteMedia(_ media: Media) {
+        cancelUpload(of: media)
```

Comment:
I like the way you've split this up into separate cancel and delete methods 👍 

Do they all need to be public, or should we make `cancelUpload(of:)` and `delete(media:)` private?

Reply:
I'm planning to use at the least the delete(media) directly to delete media.